processor/probabilisticsamplerprocessor/logsprocessor.go

```diff
@@ -75,7 +75,17 @@ func (lsp *logSamplerProcessor) processLogs(ctx context.Context, ld plog.Logs) (
 				if lidBytes == nil && lsp.samplingSource != "" {
 					if value, ok := l.Attributes().Get(lsp.samplingSource); ok {
 						tagPolicyValue = lsp.samplingSource
-						lidBytes = value.Bytes().AsRaw()
+
+						switch value.Type() {
+						case pcommon.ValueTypeStr:
```

Comment:
I could do with some input here.. I'm not sure if we should be handling string types without understanding the encoding. See PR discussion for more context.

Reply:
I would say that this doesn't matter much, as a hash will be calculated based on the value. If it's string or int, it doesn't matter, as long as they are all converted to bytes in the end. Additionally, it doesn't matter if we are using the right encoding for the string, as long as all values are byte encoded in the same way. If we were to use this information for other than just calculating a hash, it would certainly be important to properly handle the encoding.